gdnative-core/src/variant_array.rs

```diff
@@ -171,76 +156,224 @@ impl VariantArray {
 
     // pub fn bsearch(&mut self, val: (), before: bool) -> i32 {
     //     unsafe {
-    //         (get_api().godot_array_bsearch)(&mut self.0, val, before)
+    //         (get_api().godot_array_bsearch)(self.sys_mut(), val, before)
     //     }
     // }
 
     // pub fn bsearch_custom(&mut self, val: ?, obj: ?, s: ?, before: bool) -> i32 {
     //     unimplemented!();
     // }
 
+    #[doc(hidden)]
+    #[inline]
+    pub fn sys(&self) -> *const sys::godot_array {
+        &self.sys
+    }
+
+    #[doc(hidden)]
+    #[inline]
+    pub fn sys_mut(&self) -> *mut sys::godot_array {
+        &self.sys as *const _ as *mut _
+    }
+
+    #[doc(hidden)]
+    #[inline]
+    pub fn from_sys(sys: sys::godot_array) -> Self {
+        VariantArray {
+            sys,
+            _marker: PhantomData,
+        }
+    }
+}
+
+impl VariantArray<Unique> {
+    /// Creates an empty `VariantArray`.
+    #[inline]
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    #[inline]
+    pub fn into_shared(self) -> VariantArray<Shared> {
+        VariantArray::<Shared> {
+            sys: self.sys,
+            _marker: PhantomData,
+        }
+    }
```

Comment:
`self` is dropped here, but the reference isn't incremented because the constructor is used directly.